src/coreclr/vm/riscv64/stubs.cpp

```diff
@@ -1507,14 +1515,78 @@ void StubLinkerCPU::EmitCallManagedMethod(MethodDesc *pMD, BOOL fTailCall)
 //
 // Allocation of dynamic helpers
 //
+enum Riscv64_registers
+{
+    REG_ZERO,
+    REG_RA,
+    REG_SP,
+    REG_GP,
+    REG_TP,
+    REG_T0,
+    REG_T1,
+    REG_T2,
+    REG_FP,
+    REG_S1,
+    REG_A0,
+    REG_A1,
+    REG_A2,
+    REG_A3,
+    REG_A4,
+    REG_A5,
+    REG_A6,
+    REG_A7,
+    REG_S2,
+    REG_S3,
+    REG_S4,
+    REG_S5,
+    REG_S6,
+    REG_S7,
+    REG_S8,
+    REG_S9,
+    REG_S10,
+    REG_S11,
+    REG_T3,
+    REG_T4,
+    REG_T5,
+    REG_T6
+};
+
+enum Riscv64_asm_instructions
+{
+    INSTR_AUIPC = 0x17,
+    INSTR_LUI = 0x37,
+    INSTR_ADD = 0x33,
+    INSTR_ADDI = 0x13,
+    INSTR_LD = 0x3,
+    INSTR_LW = 0x3,
+    INSTR_JALR = 0x67,
+};
+
+enum Riscv64_funct3
+{
+    FUNCT3_ADD = 0,
+    FUNCT3_ADDI = 0,
+    FUNCT3_LD = 0x3,
+    FUNCT3_LW = 0x2,
+    FUNCT3_JALR = 0,
+};
```

Comment:
Could you follow the existing style in stubs.cpp?
REG is declared like `const IntReg RegSp  = IntReg(2);` in `vm/riscv64/cgencpu.h`
`instructions` and `funct3` are handled in `Emit*` functions. I think you need to update `*TypeInstr` to `Emit*` functions in this PR.

You can add registers, emit functions and `BTypeInstr`.
Thank you!

Reply:
Created `BTypeInstr` to work with the instructions of this type. But the logic of `Emit*` is unclear since it uses a call to `ITypeInstr` or others functions. Therefore, I left the current logic of functions.

Reply:
Okay. `Emit32` cannot handle instructions well for your cases.